modules/sdk-coin-tao/src/lib/iface.ts

```diff
@@ -0,0 +1,38 @@
+import { BaseTxInfo, TypeRegistry } from '@substrate/txwrapper-core/lib/types';
+import { Args } from "@substrate/txwrapper-core/lib/types/method";
+
+export { HexString } from '@polkadot/util/types';
+
+/**
+ * Method names for the transaction method. Names change based on the type of transaction e.g 'bond' for the staking transaction
+ */
+export enum MethodNames {
+  AddStake = 'addStake',
+}
+
+/**
+ * Base transaction info shared across all types of transactions
+ */
+export interface CreateBaseTxInfo {
+  baseTxInfo: BaseTxInfo;
+  options: {
+    metadataRpc: `0x${string}`;
+    registry: TypeRegistry;
+    isImmortalEra?: boolean;
+  };
+}
+
+/**
+ * Decoded TxMethod from a transaction hex
+ */
+export interface TxMethod {
+  args: AddStakeArgs;
+  name: MethodNames;
+  pallet: string;
+}
+
+export interface AddStakeArgs extends Args {
+  amount_staked: string;
```

Comment:
check if `amount_staked` or `amountStaked` i remember for DOT there was weird behaviour with naming convention when building tx